design/specs/table_inspector.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Table Inspector
+description: 
+published: true
+date: 2022-08-02
+tags: 
+editor: markdown
+dateCreated: 2022-08-02
+---
+
+## Context
+
+Users can access information and setting options for each table component using the table inspector, including columns, rows, and cells.
+
+### Design Goals
+
+- Provide easy access to table component settings.
+- Make features easier to discover.
+
+## Enabling and Disabling the Table Inspector
+
+The table inspector can be enabled or disabled manually or automatically based on event triggers.
+
+To manually enable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to show the table inspector.
+
+Some events will automatically display the table inspector:
+
+- Navigating to a newly created table after completing the data import process.
+- Navigating to a newly created table after completing the column extraction process.
+
+To disable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to hide the table inspector.
+
+### Positioning
+
+When users enable the table inspector, a panel will appear on the right side of the screen, sharing the same container as the table and pushing its contents rather than being stacked on top of it.
+
+In some cases, this will cause the table portion of the layout to have a horizontal scroll.
+
+### Inspector Modes
+
+At the top of the inspector panel is a row of tabs. Each tab includes table component-specific attributes and actions.
+
+The tabs are listed in the following order:
+
+- Table (Default)
+- Column
+- Record
+- Cell
+
+## Using the Table Inspector
+
+Selecting different inspector modes in the table inspector allows users to examine at the properties, options and settings of table components. Aside from that, various actions can also be initiated through the inspector.
+
+### Table Mode
+
+Shown when inspector mode is set to `Table`.
+
+Properties:
+
+- Table Name
+- Table Descriptions
+
+Settings:
+
+- Record Summary
+- Table Links (Constraints)
+
+Actions:
+
+- Remove Duplicates
+- Create Query from Table
+  
+![image](/assets/design/specs/table_inspector/183378847-b942ecd3-6f3c-4cd4-8cc4-3080a041b2a4.png)
+
+![image](/assets/design/specs/table_inspector/183380108-b3db8d3b-7301-4a19-bc38-cd777a4dde46.png)
+
+### Column Mode
+
+Shown when inspector mode is set to `Column`.
+
+Properties:
+
+- Column Name
+- If Column is Link:
+  - Link Source
+
+Options:
+
+- Allow NULL
+- Allow Duplicates
+- Data Type
+  - Database Options
+  - Format Options
+
+Actions:
+
+- Extract Columns to Table
+
+![image](/assets/design/specs/table_inspector/183432613-ead6315e-3802-4345-9427-820c66094797.png)
+  
+![image](/assets/design/specs/table_inspector/183433425-9a906a91-28a4-4045-9300-732af811ed8d.png)
+
+### Record Mode
```

Comment:
Can we just skip this tab for now since the only thing in there is a new feature?

Reply:
@kgodey @seancolsen @ghislaineguerin can you help me remember the conclusion on this?

Reply:
I think we were going to keep this and have a "delete record" action.